src/components/switch/switch.test.tsx

```diff
@@ -0,0 +1,418 @@
+  }).not.toThrow();
+});
+
+test("the `warning` prop should not throw an error if `loading` is true", () => {
+  expect(() => {
+    render(<Switch id="mock-input" loading warning />);
+  }).not.toThrow();
+});
+
+test("the `info` prop should not throw an error if `loading` is true", () => {
+  expect(() => {
+    render(<Switch id="mock-input" loading info />);
+  }).not.toThrow();
+});
+
+test("`helpAriaLabel` prop should set the aria-label on the Help component", () => {
+  render(<Switch label="foo" labelHelp="fooHelp" helpAriaLabel="text" />);
+
+  const help = screen.getByRole("button");
+
+  expect(help).toHaveAttribute("aria-label", "text");
+});
+
+// Required for coverage
+test("should append '(optional)' content on the label when the isOptional prop is true", () => {
+  render(<Switch isOptional label="Optional" />);
+
+  const label = screen.getByText("Optional");
+  const labelContainer = screen.getByTestId("label-container");
+
+  expect(label).toHaveTextContent("Optional");
+  expect(labelContainer).toHaveStyleRule("content", '"(optional)"', {
+    modifier: "::after",
+  });
+});
+
+// Required for coverage
+test("the correct border colour is applied when `error` validation is true", () => {
+  render(<Switch error />);
+
+  const switchSlider = screen.getByTestId("slider");
+
+  expect(switchSlider).toHaveStyleRule(
+    "border-color: var(--colorsSemanticNegative500)"
+  );
+});
+
+// Required for coverage
+test("the correct border colour is applied when `warning` validation is true", () => {
+  render(<Switch warning />);
+
+  const switchSlider = screen.getByTestId("slider");
+
+  expect(switchSlider).toHaveStyleRule(
+    "border-color: var(--colorsSemanticCaution500)"
+  );
+});
+
+// Required for coverage
+test("the correct background colour is applied to the `ErrorBorder` element when `error` validation is a string", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch error="this is an error" />
+    </CarbonProvider>
+  );
+
+  const switchSlider = screen.getByTestId("error-border");
+
+  expect(switchSlider).toHaveStyleRule(
+    "background-color: var(--colorsSemanticNegative500)"
+  );
+});
+
+// Required for coverage
+test("the correct background colour is applied to the `ErrorBorder` element when `warning` validation is a string", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch warning="this is a warning" />
+    </CarbonProvider>
+  );
+
+  const switchSlider = screen.getByTestId("error-border");
+
+  expect(switchSlider).toHaveStyleRule(
+    "background-color: var(--colorsSemanticCaution500)"
+  );
+});
+
+// Required for coverage
+test("renders the expected hint text when `labelHelp` is a string value", () => {
+  render(
+    <CarbonProvider validationRedesignOptIn>
+      <Switch label="foo" labelHelp="hint text" warning="this is a warning" />
+    </CarbonProvider>
+  );
+
+  expect(screen.getByTestId("hint-text")).toBeVisible();
+});
```

Comment:
**suggestion (non-blocking):** Aware that this is for coverage so happy for this to be left as it is however, I think we could slightly amend the description to make it a bit more clear what is being tested. I also think we could also just assert that the text is visible instead of using the test-id here
```suggestion
test("renders `labelHelp` as hint text when `validationRedesignOptIn` flag is true", () => {
  render(
    <CarbonProvider validationRedesignOptIn>
      <Switch label="foo" labelHelp="hint text" warning="this is a warning" />
    </CarbonProvider>
  );

  expect(screen.getByText("hint text")).toBeVisible();
});
```